`MigrateProcessInstanceConcurrentNoBatchingTest` is flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? at least [once](https://github.com/camunda/camunda/actions/runs/10405955810/job/28817973356)
- Does it block your work? it may block merges
- Do we suspect that it is a real failure? no, it's a concurrent modification in rocksdb by the test

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Error:  io.camunda.zeebe.engine.processing.processinstance.migration.MigrateProcessInstanceConcurrentNoBatchingTest.shouldRejectMigrationForActiveCatchEventAttachedToEventBasedGateway -- Time elapsed: 0.120 s <<< ERROR!
java.lang.RuntimeException: Unexpected error occurred during RocksDB transaction.
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction(DefaultTransactionContext.java:42)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.ensureInOpenTransaction(TransactionalColumnFamily.java:343)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.get(TransactionalColumnFamily.java:139)
	at io.camunda.zeebe.stream.impl.state.NextValueManager.getCurrentValue(NextValueManager.java:63)
	at io.camunda.zeebe.stream.impl.state.NextValueManager.getCurrentValue(NextValueManager.java:54)
	at io.camunda.zeebe.stream.impl.state.NextValueManager.getNextValue(NextValueManager.java:38)
	at io.camunda.zeebe.stream.impl.state.DbKeyGenerator.nextKey(DbKeyGenerator.java:41)
	at io.camunda.zeebe.engine.processing.processinstance.migration.MigrateProcessInstanceConcurrentNoBatchingTest.shouldRejectMigrationForActiveCatchEventAttachedToEventBasedGateway(MigrateProcessInstanceConcurrentNoBatchingTest.java:982)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caused by: org.rocksdb.RocksDBException: WriteBatch has wrong count
	at org.rocksdb.Transaction.commit(Native Method)
	at org.rocksdb.Transaction.commit(Transaction.java:216)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.commitInternal(ZeebeTransaction.java:147)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInNewTransaction(DefaultTransactionContext.java:62)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction(DefaultTransactionContext.java:33)
	... 8 more
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

The test case modifies RocksDB by asking for a new key from the keygenerator.

https://github.com/camunda/camunda/blob/d7f9b87f79454b43beb59eb2b685f40224336552/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateProcessInstanceConcurrentNoBatchingTest.java#L979-L982

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateProcessInstanceConcurrentNoBatchingTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateProcessInstanceConcurrentNoBatchingTest.java
@@ -977,6 +977,8 @@ public class MigrateProcessInstanceConcurrentNoBatchingTest {
         .await();
 
     // we generate a key for the intermediate catch event so we can activate it using events
+    // this requires us to pause processing to avoid concurrent database transaction modification
+    ENGINE.pauseProcessing(1);
     final var keyGenerator =
         ((MutableProcessingState) ENGINE.getProcessingState()).getKeyGenerator();
     final var intermediateCatchEventKey = keyGenerator.nextKey();

